rinja_parser/src/target.rs

```diff
@@ -195,19 +227,29 @@ fn collect_targets<'a, T>(
     Ok((i, (singleton, targets)))
 }
 
-fn only_one_rest_pattern(targets: Vec<Target<'_>>) -> Result<Vec<Target<'_>>, ParseErr<'_>> {
-    let snd_wildcard = targets
-        .iter()
-        .filter_map(|t| match t {
-            Target::Rest(s) => Some(s),
-            _ => None,
-        })
-        .nth(1);
-    if let Some(snd_wildcard) = snd_wildcard {
-        return Err(nom::Err::Failure(ErrorContext::new(
-            "`..` can only be used once per tuple pattern",
-            snd_wildcard,
-        )));
+fn only_one_rest_pattern<'a>(
+    targets: Vec<Target<'a>>,
+    allow_named_rest: bool,
+    type_kind: &str,
+) -> Result<Vec<Target<'a>>, ParseErr<'a>> {
+    let mut found_rest = false;
+
+    for target in &targets {
+        if let Target::Rest(s) = target {
+            if !allow_named_rest && s.inner.is_some() {
+                return Err(nom::Err::Failure(ErrorContext::new(
+                    format!("`@ ..` cannot be used in {type_kind}"),
```

Comment:
Rust uses "`@ ..` is only allowed in slice patterns". I like that better, because it tells you what you can do, instead of [what you can't do](https://lostpedia.fandom.com/wiki/Don%27t_tell_me_what_I_can%27t_do).

The code looks fine, so if you like the current phrasing better, then it's not a showstopper for me!

Reply:
No you're right, I prefer your version. Updating!